modules/@apostrophecms/notification/ui/apos/components/AposNotification.vue

```diff
@@ -50,157 +51,117 @@
-      const textContent = div.textContent || div.innerText || '';
-      if (textContent.length > 160) {
-        classes.push('apos-notification--long');
-      }
+  if (Array.isArray(props.notification.classes) && props.notification.classes.length) {
+    classes.push(...props.notification.classes);
+  }
 
-      return classes.join(' ');
-    },
-    iconComponent() {
-      if (this.notification.icon) {
-        return this.notification.icon;
-      } else {
-        return 'circle-icon';
-      }
-    }
-  },
-  async mounted() {
-    if (this.notification.dismiss) {
-      setTimeout(() => {
-        this.$emit('close', this.notification._id);
-      }, 1000 * this.notification.dismiss);
-    }
-    this.$refs.label.addEventListener('click', (e) => {
-      if (e.target.hasAttribute('data-apos-bus-event')) {
-        this.close();
-      }
-    });
-
-    if (this.job) {
-      try {
-        const {
-          total,
-          processed,
-          percentage
-        } = await apos.http.get(this.job.route, {});
-
-        this.job.total = total;
-        this.job.processed = processed || 0;
-        this.job.percentage = percentage;
-        this.job.ids = this.notification.job.ids || [];
-
-        await this.pollJob();
-      } catch (error) {
-        console.error('Unable to find notification job:', this.notification.job._id);
-        this.job = null;
-      }
-    }
-    // Notifications may include events to emit.
-    if (this.notification.event?.name) {
-      try {
-        // Clear the event to make sure it's only emitted once across browsers.
-        const safe = await this.clearEvent(this.notification._id);
-
-        if (safe) {
-          // The notification doc will only still have the event in one instance.
-          apos.bus.$emit(this.notification.event.name, this.notification.event.data);
-        }
-      } catch (error) {
-        console.error(this.$t('apostrophe:notificationClearEventError'));
-      }
-    }
-  },
-  methods: {
-    close() {
-      this.$emit('close', this.notification._id);
-    },
-    localize(s) {
-      let result;
-      if (this.notification.localize !== false) {
-        result = this.$t(s, this.notification.interpolate || {});
-      } else {
-        // Any interpolation was done before insertion
-        result = s;
-      }
-      return result;
-    },
-    async pollJob() {
-      if (!this.job?.total) {
-        return;
-      }
-      const job = await apos.http.get(this.job.route, {});
-      this.job.processed = job.processed;
-      this.job.percentage = job.percentage;
-
-      if (this.job.processed < this.job.total && !job.ended) {
-        await new Promise(resolve => {
-          setTimeout(resolve, 500);
-        });
-
-        await this.pollJob();
-      } else {
-        if (this.job.ids) {
-          apos.bus.$emit('content-changed', {
-            docIds: this.job.ids,
-            action: this.job.action || 'batch-update'
-          });
-        }
```

Comment:
Sorry moved this component to composition, cleaner to work with store. and also I wanted since a tiny component. Globally work the same. Has been decoupled from jobs since more global logic exist in pinia notification store.